dev.Dockerfile

```diff
@@ -41,6 +43,11 @@ ARG NO_UI
 # Install dependencies and install/build lightning-terminal.
 RUN apk add --no-cache --update alpine-sdk make \
   && cd /go/src/github.com/lightninglabs/lightning-terminal \
+  # If a custom lnd version is supplied, force it now.
+  && if [ -n "$LND_VERSION" ]; then \
+    go get -v github.com/lightningnetwork/lnd@$LND_VERSION \
```

Comment:
This should work. Keep in mind though that the lnd version and other daemons included in litd are quite tightly coupled. So you won't be able to have big version differences without running into compilation errors. But for quickly testing out a side branch this works.